### PR TITLE
Generalize prepbufr adpsfc converter

### LINKED ISSUE #1299: Generalize the NCEP PREPBUFR ADPSFC to IODA converter
## Description
The new `prepbufr_adpsfc_api.py` converter is currently set up to use the test data only. This issue would add the ability to call the converter using any input, output, and date while retaining the test data as the default

## Requirements

Converter produces correct output for files given when called with arguments and produces original output using test data when called without arguments.

## Acceptance Criteria (Definition of Done)
Converter produces expected output in all situations

## Dependencies
none

### PR DESCRIPTION
**Pull request by haydenlj**
## Description

Proposed changes to generalize the new NCEP PREPBUFR ADPSFC to IODA converter so that input and output files as well as the date can be passed from the command line

## Issue(s) addressed

Resolves #1299 

## Dependencies

none

## Impact
none

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have run the unit tests before creating the PR

### CODE REVIEW

File: test/testinput/prepbufr_adpsfc_api.py [hunk 1]
```diff
@@ -210,5 +212,24 @@ def test_bufr_to_ioda():
...
+    optional.add_argument('-f', '--filenames', type=str, default=DATA_PATH,
+                          dest='filenames', help='adpsfc file name')
+    optional.add_argument('-o', '--output', type=str, default=OUTPUT_PATH,
+                          dest='output', help='output filename')
+    optional.add_argument('-d', '--date', type=str, default='2021 08 01 00 00',
+                          dest='date', help='enable verbose debug messages')
+
+    args = parser.parse_args()
+
+    test_bufr_to_ioda(args.filenames, args.output, args.date)
```
Comment (fabiolrdiniz) [inline test/testinput/prepbufr_adpsfc_api.py:234]:
```suggestion
    optional.add_argument('-f', '--filename', type=str, default=DATA_PATH,
                          dest='filename', help='adpsfc file name')
    optional.add_argument('-o', '--output', type=str, default=OUTPUT_PATH,
                          dest='output', help='output filename')
    optional.add_argument('-d', '--date', type=str, default='2021 08 01 00 00',
                          dest='date', metadatavar='YYYY mm dd HH MM', help='enable verbose debug messages')

    args = parser.parse_args()

    test_bufr_to_ioda(args.filename, args.output, args.date)
```

File: test/testinput/prepbufr_adpsfc_api.py [hunk 2]
```diff
@@ -53,7 +52,7 @@ def test_bufr_to_ioda():
    print("Get time")
    dhr = r.get('obsTimeMinusCycleTime') #Needs to be converted to seconds since Epoch time from [-3,3]
    print("cycleTimeSinceEpoch") #For now, file time is put in manually 
-   cycleTimeSinceEpoch = np.int64(calendar.timegm(time.strptime('2021 08 01 00 00', '%Y %m %d %H %M')))
+   cycleTimeSinceEpoch = np.int64(calendar.timegm(time.strptime(date, '%Y%m%d%H%M')))
```
Comment (fabiolrdiniz) [inline test/testinput/prepbufr_adpsfc_api.py:55]: Don't need the spaces in the format?
```suggestion
   cycleTimeSinceEpoch = np.int64(calendar.timegm(time.strptime(date, '%Y %m %d %H %M')))
```

File: test/testinput/prepbufr_adpsfc_api.py [hunk 3]
```diff
@@ -210,5 +212,24 @@ def test_bufr_to_ioda():
...
+    DATA_PATH = './testinput/gdas.t12z.adpsfc.prepbufr'
+    OUTPUT_PATH = './testrun/prepbufr_adpsfc_api.nc'
+
+    optional = parser.add_argument_group(title='optional arguments')
+    optional.add_argument('-f', '--filename', type=str, default=DATA_PATH,
+                          dest='filename', help='adpsfc file name')
+    optional.add_argument('-o', '--output', type=str, default=OUTPUT_PATH,
+                          dest='output', help='output filename')
+    optional.add_argument('-d', '--date', type=str, default='2021 08 01 00 00',
+                          dest='date', metadatavar='YYYY mm dd HH MM', help='enable verbose debug messages')
```
Comment (fabiolrdiniz) [inline test/testinput/prepbufr_adpsfc_api.py:230]: Sorry, I was editing my suggestion above while you were adding them to the branch.
```suggestion
                          dest='date', metavar='YYYY mm dd HH MM', help='analysis cycle date'
```

**Review by fabiolrdiniz — APPROVED** [review]
Thanks, @haydenlj!

**Review by BenjaminRuston — APPROVED** [review]
thank you @haydenlj

**Review by PatNichols — APPROVED** [review]

File: test/testinput/prepbufr_adpsfc_api.py [hunk 4]
```diff
@@ -210,5 +212,24 @@ def test_bufr_to_ioda():
...
+
+    DATA_PATH = './testinput/gdas.t12z.adpsfc.prepbufr'
+    OUTPUT_PATH = './testrun/prepbufr_adpsfc_api.nc'
+
+    optional = parser.add_argument_group(title='optional arguments')
+    optional.add_argument('-f', '--filename', type=str, default=DATA_PATH,
+                          dest='filename', help='adpsfc file name')
+    optional.add_argument('-o', '--output', type=str, default=OUTPUT_PATH,
+                          dest='output', help='output filename')
+    optional.add_argument('-d', '--date', type=str, default='202108010000',
```
Comment (BenjaminRuston) [inline test/testinput/prepbufr_adpsfc_api.py:229]: @emilyhcliu and @nicholasesposito ,,,, note this default time does not match the `t12z` in the sample file shown in `DATA_PATH` may want to make these consistent

Comment (nicholasesposito) [reply test/testinput/prepbufr_adpsfc_api.py:229]: Oh, right, I didn't think about this. So the adpsfc file that was already in here was this 12z file (I don't remember from when). I didn't want to change the other results so I kept this file and used the converter on it, but apparently forgot to change the information in the python script. 

Do we prefer to use the 12z file that's already there or the 202108010000 file?

**Pull request merged**